gcc/rust/typecheck/rust-hir-trait-ref.h

```diff
@@ -149,8 +148,7 @@ class TraitItemReference
     return new TyTy::ErrorType (get_mappings ().get_hirid ());
   }
 
-  TyTy::BaseType *get_type_from_typealias (/*const*/
-					   HIR::TraitItemType &type) const;
+  TyTy::BaseType *get_type_from_typealias () const;
```

Comment:
This isn't right we mght need to have the TraitItemType to do resolution of the associated type properly when it comes to handling opaque types and generic argument binding which still needs to be done.